CS3219-AY2324S1/ay2324s1-course-assessment-g03
# feat: add admin questions portal

![localhost_8000_admin_questions (1)](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g03/assets/59964533/e3d34c3b-55c3-4b33-8856-9e799e99b1dc)

### Changes
- [x] `/questions` page
  - [x] search by title or description
  - [x] filters (difficulty, category, paid status, topics)  
  - [x] sort by column (toggle ascending or descending, default id ascending)
  - [x] row actions (details, edit, delete)
  - [x] create question modal
  - [x] update question modal
  - [x] delete question modal 
- [x] `/questions/:questionId` details page
- [x] update `react-query` to `tanstack/react-query`

### How to test
1. All authenticated users should be able to view questions via the new **Questions** nav link
For admin actions:
2. Connect to the local `user-service-db` instance via MongoDB Compass (connection URL: `mongodb://localhost:27018/user-service?retryWrites=true&w=majority`) or Prisma Studio
3. Click into the `users_db` database and `User` collection
4. Edit your account entry's `role` to be `ADMIN`
5. You should see create, update and delete options appear on the questions page
